api/type.rb

```diff
@@ -166,15 +166,23 @@ def initialize(name)
       NAME = Api::Type::String.new('name')
     end
 
+    # Properties that are fetched externally
+    class FetchedExternal < Type
+      attr_writer :resource
+
+      def api_name
+        name
+      end
+    end
+
     # Represents a fingerprint.  A fingerprint is an output-only
     # field used for optimistic locking during updates.
-    class Fingerprint < String
+    class Fingerprint < FetchedExternal
       def validate
         super
         @output = true if @output.nil?
         # TODO(ndmckinley): This doesn't work in puppet, chef, or ansible.
```

Comment:
If you're removing the line of code this comment refers to, can you also remove the comment?